### PR TITLE
Disable spark dependencies for self provisioned es

### LINKED ISSUE #316: Disable dependnecies for self provisioned ES
Spark dependencies does not support TLS #294  so we cannot use them with self-provisioned ES.

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #316 

We don't want auto-enabled spark deps for self-provisioned ES

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

**Comment by jpkrohling** [comment]
This change is [Reviewable](https://reviewable.io/reviews/jaegertracing/jaeger-operator/319)

**Comment by objectiser** [comment]
@pavolloffay Do we need to document this somewhere, or produce a warning message?

**Comment by pavolloffay** [comment]
The log message in the controller would be logged on every loop. I will just add it to the readme
```
INFO[0055] Auto enabling for Spark dependencies is disabled. Spark dependencies are not support for self provisioned Elasticsearch  instance=simple-prod namespace=openshift-logging
INFO[0088] Auto enabling for Spark dependencies is disabled. Spark dependencies are not support for self provisioned Elasticsearch  instance=simple-prod namespace=openshift-logging
INFO[0103] Auto enabling for Spark dependencies is disabled. Spark dependencies are not support for self provisioned Elasticsearch  instance=simple-prod namespace=openshift-logging
INFO[0108] Auto enabling for Spark dependencies is disabled. Spark dependencies are not support for self provisioned Elasticsearch  instance=simple-prod namespace=openshift-logging
INFO[0113] Auto enabling for Spark dependencies is disabled. Spark dependencies are not support for self provisioned Elasticsearch  instance=simple-prod namespace=openshift-logging
INFO[0118] Auto enabling for Spark dependencies is disabled. Spark dependencies are not support for self provisioned Elasticsearch  instance=simple-prod namespace=openshift-logging
INFO[0123] Auto enabling for Spark dependencies is disabled. Spark dependencies are not support for self provisioned Elasticsearch  instance=simple-prod namespace=openshift-logging
INFO[0128] Auto enabling for Spark dependencies is disabled. Spark dependencies are not support for self provisioned Elasticsearch  instance=simple-prod namespace=openshift-logging
INFO[0134] Auto enabling for Spark dependencies is disabled. Spark dependencies are not support for self provisioned Elasticsearch  instance=simple-prod namespace=openshift-logging
INFO[0139] Auto enabling for Spark dependencies is disabled. Spark dependencies are not support for self provisioned Elasticsearch  instance=simple-prod namespace=openshift-logging
INFO[0144] Auto enabling for Spark dependencies is disabled. Spark dependencies are not support for self provisioned Elasticsearch  instance=simple-prod namespace=openshift-logging
INFO[0149] Auto enabling for Spark dependencies is disabled. Spark dependencies are not support for self provisioned Elasticsearch  instance=simple-prod namespace=openshift-logging
INFO[0154] Auto enabling for Spark dependencies is disabled. Spark dependencies are not support for self provisioned Elasticsearch  instance=simple-prod namespace=openshift-logging
```

File: README.adoc [hunk 1]
```diff
@@ -260,7 +260,7 @@ spec:
 
 Under some circumstances, the Jaeger Operator can make use of the link:https://github.com/openshift/elasticsearch-operator[Elasticsearch Operator] to provision a suitable Elasticsearch cluster.
 
-IMPORTANT: this feature is experimental and has been tested only on OpenShift clusters. Elasticsearch also requires the memory setting to be configured like `minishift ssh -- 'sudo sysctl -w vm.max_map_count=262144'`
+IMPORTANT: this feature is experimental and has been tested only on OpenShift clusters. Elasticsearch also requires the memory setting to be configured like `minishift ssh -- 'sudo sysctl -w vm.max_map_count=262144'`. Spark dependencies are neither supported link:https://github.com/jaegertracing/jaeger-operator/issues/294[#294].
```
Comment (objectiser) [inline README.adoc]: "Spark dependencies are not support with this feature link:...."

**Pull request merged**